xc7/utils/prjxray_form_channels.py

```diff
@@ -676,13 +687,106 @@ def form_tracks(conn):
   """, (node,)):
                 unique_pos.add((grid_x, grid_y))
 
-            xs, ys = points.decompose_points_into_tracks(unique_pos)
-            tracks_list, track_connections = tracks.make_tracks(xs, ys, unique_pos)
-            tracks_model = tracks.Tracks(tracks_list, track_connections)
+            tracks_to_insert.append(create_track(node, unique_pos))
+
+    # Create constant tracks
+    vcc_track_to_insert, gnd_track_to_insert = create_constant_tracks(conn)
+    vcc_idx = len(tracks_to_insert)
+    tracks_to_insert.append(vcc_track_to_insert)
+    gnd_idx = len(tracks_to_insert)
+    tracks_to_insert.append(gnd_track_to_insert)
+
+    track_pkeys = insert_tracks(conn, tracks_to_insert)
+    vcc_track_pkey = track_pkeys[vcc_idx]
+    gnd_track_pkey = track_pkeys[gnd_idx]
+
+    c.execute("""
+INSERT INTO constant_sources(vcc_track_pkey, gnd_track_pkey) VALUES (?, ?)
+        """, (vcc_track_pkey, gnd_track_pkey,))
+
+    conn.commit()
+
```

Comment:
Maybe split the stuff below into its own function?
```suggestion
    convert_hardpins_to_constant_network(....)


def convert_hardpins_to_constant_network(...):
```